engine/src/main/java/com/datatorrent/stram/plan/physical/PhysicalPlan.java

```diff
@@ -344,9 +346,67 @@ public PhysicalPlan(LogicalPlan dag, PlanContext ctx) {
         addLogicalOperator(n);
       }
     }
+    
+    inlinePrefs.prefs.clear();
+    localityPrefs.prefs.clear();
+    
+    // Add inlinePrefs and localityPreds for affinity rules
```

Comment:
Great point. Did not handle it earlier. 
Removed clearing of inlinePrefs and localityPrefs. These preference are applied again on dynamic plan changes. So affinity will be set again on dynamic changes. As for anti-affinity, added population of antiAffinityPreferences for containers on re-allocation.

Added a test case for the same.